parlai/core/teachers.py

```diff
@@ -2377,32 +2416,46 @@ def _enqueue_request(self):
         """
         Queue a request for loading to the data loader.
         """
-        self.data_loader.request_load(self.receive_data, self.get_chunk, ())
+        if self.threading:
+            self.data_loader.request_load(self.receive_data, self.get_chunk, ())
+        else:
+            self.receive_data(self.get_chunk(), direct_result=True)
 
-    def receive_data(self, future):
+    def receive_data(self, future, direct_result=False):
```

Comment:
Hmm maybe it'd be worth adding a line of documentation on `future`? I understand from the line above that it's a chunk, but offhand I don't even know what a chunk is...